libs/php-commons/XSSHelpers.php

```diff
@@ -57,179 +65,180 @@ public static function xscrub($data)
+                    // End IE 11 workarounds
+                    // Start table related support
+                    case 'col':
+                        $pattern[] = '/&lt;col style=&(quot|#039);(.+)&(quot|#039); width=&(quot|#039);(.+)&(quot|#039); span=&(quot|#039);(.+)&(quot|#039);(\s.+)?&gt;/Ui';
+                        $replace[] = '<col style="\2" width="\5" span="\8">';
+
+                        break;
+                    case 'td':
+                        $pattern[] = '/&lt;td(\s.+)?&gt;/U';
+                        $replace[] = '<td>';
+                        $pattern[] = '/&lt;\/td(\s.+)?&gt;/U';
+                        $replace[] = '</td>';
+
+                        break;
+                    // End table related support
+                    default:
+                        $pattern[] = '/&lt;(\/)?' . $tag . '(\s.+)?&gt;/U';
+                        $replace[] = '<\1' . $tag . '>';
+
+                        break;
+                }
             }
-        }
-        while ($in != html_entity_decode($in, ENT_QUOTES | ENT_HTML5, $encoding))
-        {
-            $in = html_entity_decode($in, ENT_QUOTES | ENT_HTML5, $encoding);
-        }
-        $in = preg_replace(self::$specialPattern, self::$specialReplace, $in); // modifiers to support features
+            while ($in != html_entity_decode($in, ENT_QUOTES | ENT_HTML5, $encoding))
+            {
+                $in = html_entity_decode($in, ENT_QUOTES | ENT_HTML5, $encoding);
+            }
+            $in = preg_replace(self::$specialPattern, self::$specialReplace, $in); // modifiers to support features
 
-        $in = preg_replace($pattern, $replace, htmlspecialchars($in, ENT_QUOTES, $encoding));
+            $in = preg_replace($pattern, $replace, htmlspecialchars($in, ENT_QUOTES, $encoding));
 
-        // verify tag grammar
-        $matches = array();
-        preg_match_all('/\<(\/)?([A-Za-z]+)(\s.+)?\>/U', $in, $matches, PREG_PATTERN_ORDER);
-        $openTags = array();
-        $numTags = count($matches[2]);
-        for ($i = 0; $i < $numTags; $i++)
-        {
-            if ($matches[2][$i] != 'br'
-                && $matches[2][$i] != 'img'
-                && $matches[2][$i] != 'col')
+            // verify tag grammar
+            $matches = array();
+            preg_match_all('/\<(\/)?([A-Za-z]+)(\s.+)?\>/U', $in, $matches, PREG_PATTERN_ORDER);
+            $openTags = array();
+            $numTags = count($matches[2]);
+            for ($i = 0; $i < $numTags; $i++)
             {
-                //echo "examining: {$matches[1][$i]}{$matches[2][$i]}\n";
-                // proper closure
-                if ($matches[1][$i] == '/' && isset($openTags[$matches[2][$i]]) && $openTags[$matches[2][$i]] > 0)
-                {
-                    $openTags[$matches[2][$i]]--;
-                // echo "proper\n";
-                }
-                // new open tag
-                else
+                if ($matches[2][$i] != 'br'
```

Comment:
Might want to consider changing this to an in-array conditional check.  You're checking the same value against three different strings.  a list that could conceivably grow.  And do you need to check/ensure case here?